core/store/src/db.rs

```diff
@@ -137,6 +137,14 @@ pub trait Database: Sync + Send {
     /// This is a no-op for in-memory databases.
     fn flush(&self) -> io::Result<()>;
 
+    /// Compact database representation.
+    ///
+    /// If the database supports it a form of compaction, calling this function
+    /// is blocking until compaction finishes. Otherwise, this is a no-op.
+    fn compact(&self) -> io::Result<()> {
+        Ok(())
+    }
```

Comment:
Very minor, but I'd `;` here, to make sure that implementors are explicit about "no, compaction doesn't make sense for me"

Reply:
done